pkg/odo/cli/component/devfile.go

```diff
@@ -181,6 +181,36 @@ func (do *DeleteOptions) DevfileComponentDelete() error {
 	return nil
 }
 
+// RunTestCommand runs the specific command in devfile
+func (to *TestOptions) RunTestCommand() error {
+	// Parse devfile
+	devObj, err := parser.ParseAndValidate(to.devfilePath)
+	if err != nil {
+		return err
+	}
+
+	componentName, err := getComponentName(to.componentContext)
+	if err != nil {
+		return err
+	}
+
+	var platformContext interface{}
+	if pushtarget.IsPushTargetDocker() {
+		platformContext = nil
+	} else {
+		kc := kubernetes.KubernetesContext{
+			Namespace: to.KClient.Namespace,
+		}
+		platformContext = kc
+	}
+
+	devfileHandler, err := adapters.NewComponentAdapter(componentName, to.componentContext, devObj, platformContext)
+	if err != nil {
+		return err
+	}
+	return devfileHandler.Test(to.testCommand, true)
```

Comment:
we should not hardcode `show` to `true`, get it from the test cli show flag

Reply:
Updated